src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java

```diff
@@ -0,0 +1,210 @@
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.mail.*;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.search.SearchTerm;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.String.format;
+
+public class EmailFixture extends SlimFixture {
+    private final static Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private static final Date HOURS_BACK = new Date(System.currentTimeMillis() - 1000 * 60 * 60 * 24);
+    private Store store;
+
+    public EmailFixture() {
+
+    }
+
+    /**
+     * | set mail provider with host | <i>host</i> | port | <i>port</i> | user | <i>username</i> | password | <i>password</i> |
+     */
+    public void setMailProviderWithHostPortUserPassword(final String host, final String port, final String username, final String password) {
+        final Properties props = new Properties();
+        try {
+            final Session session = Session.getDefaultInstance(props, null);
+            store = session.getStore("imaps");
+            store.connect(host, username, password);
+        } catch (final MessagingException e) {
+            throw new StopTestException("Cannot connect to mailserver");
+        }
+    }
+
+    /**
+     * | ensure | mail received with subject | <i>subject</i> | from | <i>receiver</i> |
+     *
+     * @param subject
+     * @return
+     */
+    public String mailReceivedWithSubjectFrom(final String subject, final String receiver) {
+        return getMailText(new SearchParameters(subject, receiver, HOURS_BACK));
+    }
+
+    /**
+     * | $result= | extract text with | <i>regex</i> | from mail with | <i>subject</i> | from | <i>receiver</i> |
+     */
+    public String extractTextWithFromMailWithFrom(final String regex, final String subject, final String receiver) {
+        final String text = getMailText(new SearchParameters(subject, receiver, HOURS_BACK));
+        return getText(text, regex);
+    }
+
+    private String getText(final String text, final String regexpattern) {
+        LOGGER.debug("Text found: " + text);
+        LOGGER.debug("Regex: " + regexpattern);
+        final Pattern pattern = Pattern.compile(regexpattern);
+        final Matcher matcher = pattern.matcher(text);
+        final boolean isMatch = matcher.find();
+        if (isMatch) {
+            return matcher.group(1);
+        }
+        throw new SlimFixtureException(false, format("No match found in text %s with regex %s", text, regexpattern));
+    }
+
+    private String getMailText(SearchParameters params) {
+        try {
+            final Folder inbox = getInboxFolder();
+            Message msg = getRecentMessagesMatching(inbox, params);
+            return getBody(msg);
+        } catch (final Exception ex) {
```

Comment:
If a SlimFixtureException is thrown I would not create a new one, but just rethrow the existing